angular-workspace/projects/ni/nimble-angular/table-column/anchor/nimble-table-column-anchor-router-link-with-href.directive.ts

```diff
@@ -28,11 +31,54 @@ export class NimbleTableColumnAnchorRouterLinkWithHrefDirective extends RouterLi
             return;
         }
 
-        // Let the router handle this navigation
-        super.routerLink = href;
-        if (!this.onClick(clickEvent.button, clickEvent.ctrlKey, clickEvent.shiftKey, clickEvent.altKey, clickEvent.metaKey)) {
+        if (!this.handleLinkClick(href, clickEvent.button, clickEvent.ctrlKey, clickEvent.shiftKey, clickEvent.altKey, clickEvent.metaKey)) {
             clickEvent.preventDefault();
         }
-        super.routerLink = null;
+    }
+
+    private handleLinkClick(url: string, button: number, ctrlKey: boolean, shiftKey: boolean, altKey: boolean, metaKey: boolean): boolean {
+        // eslint-disable-next-line @typescript-eslint/dot-notation
+        const router = this['router'] as Router | null;
+        // eslint-disable-next-line @typescript-eslint/dot-notation
+        const locationStrategy = this['locationStrategy'] as LocationStrategy;
+
+        if (!(locationStrategy instanceof HashLocationStrategy) && router !== null) {
+            // Only attempt routing in apps not using hash-based routing
+            let routerUrl = url;
+            const baseHref = locationStrategy.getBaseHref();
+            if (!baseHref || this.urlStartsWithBaseHref(routerUrl, baseHref)) {
+                // Let the router handle this navigation
+                routerUrl = this.removeBaseHrefFromUrl(routerUrl, baseHref);
```

Comment:
I think I need more explanation of what is going on here. Maybe an offline discussion. But all this logic of ripping out chunks of the URL and only supporting certain routing seems super finicky. URLs can have lots of formats and structures, it's not clear to me how what's being shown is a good narrow / generic solution

Reply:
Ultimately, we want to be able to adapt a user-provided URL so that it can work with the router _or_ the default link click hander. The router wants URLs that are relative to the base href. The default link click handler wants full URLs.

Our approach is to let the router do the navigation if the URL starts with the base href (including the degenerate case where there is no base href). In that case, we remove the base href from the URL so that the router is happy. Otherwise, we let the default link click handler do the navigation on the URL as-is.

As far as why we don't attempt router navigation when the app is using hash-based routing, I'm following what we do in the SLE navigation tree. I think it doesn't make sense for hash-based routing, though I'm fuzzy on this.